app/src/main/java/com/hedvig/app/feature/home/ui/changeaddress/ChangeAddressResultActivity.kt

```diff
@@ -0,0 +1,127 @@
+        setContent {
+            ChangeAddressResultScreen(
+                result = intent.getParcelableExtra(RESULT)!!,
+                navigateToHome = { startActivity(LoggedInActivity.newInstance(this, withoutHistory = true)) },
+                navigateToChat = { startActivity(ChatActivity.newInstance(this)) }
+            )
+        }
+    }
+
+    sealed class Result : Parcelable {
+        @Parcelize
+        data class Success(
+            val date: LocalDate,
+        ) : Result()
+
+        @Parcelize
+        object UnderwritingLimitsHit : Result()
+    }
+
+    companion object {
+        private const val RESULT = "RESULT"
+
+        fun newInstance(context: Context, result: Result) = Intent(
+            context,
+            ChangeAddressResultActivity::class.java
+        ).apply {
+            putExtra(RESULT, result)
+        }
+    }
+}
+
+@Composable
+fun ChangeAddressResultScreen(
+    result: ChangeAddressResultActivity.Result,
+    navigateToHome: () -> Unit,
+    navigateToChat: () -> Unit,
+) {
+    MdcTheme {
+        Surface(color = MaterialTheme.colors.background) {
+            when (result) {
+                is ChangeAddressResultActivity.Result.Success -> {
+                    ITPB(
+                        illustration = painterResource(R.drawable.illustration_move_success),
+                        title = stringResource(R.string.moving_confirmation_success_title),
+                        paragraph = stringResource(
+                            R.string.moving_confirmation_success_paragraph,
+                            DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT).format(result.date)
+                        ),
+                        onClick = navigateToHome,
+                        buttonContent = {
+                            Text(text = stringResource(R.string.moving_confirmation_success_button_text))
+                        }
+                    )
+                }
+                ChangeAddressResultActivity.Result.UnderwritingLimitsHit -> {
+                    ITPB(
+                        illustration = painterResource(R.drawable.illustration_helicopter),
+                        title = stringResource(R.string.moving_uw_failure_title),
+                        paragraph = stringResource(R.string.moving_uw_failure_paragraph),
+                        onClick = navigateToChat,
+                        buttonContent = {
+                            Icon(
+                                painter = painterResource(R.drawable.ic_chat_white),
+                                contentDescription = null,
+                                modifier = Modifier.size(ButtonDefaults.IconSize)
+                            )
+                            Spacer(Modifier.size(ButtonDefaults.IconSpacing))
+                            Text(stringResource(R.string.moving_uw_failure_button_text))
+                        }
+                    )
+                }
+            }
+        }
+    }
+}
+
```

Comment:
Are there any drawbacks for having previews in the activity?

Reply:
I'm not sure, to be honest. Probably not?

I guess something that could be worth investigating is what the compose compiler does with Previews. Preferably, they should be stripped in release targets, right? 🤔

Reply:
Yeah exactly. But it probably makes sense to have the previews in the same file since they are so tightly coupled with the actual view.